probe-rs/src/probe/usb_util.rs

```diff
@@ -26,11 +34,23 @@ impl InterfaceExt for Interface {
 
     fn read_bulk(&self, endpoint: u8, buf: &mut [u8], timeout: Duration) -> io::Result<usize> {
         let fut = async {
-            let comp = self.bulk_in(endpoint, RequestBuffer::new(buf.len())).await;
+            let mut ep_in = self.endpoint::<Bulk, In>(endpoint).unwrap();
+            let transfer = ep_in.allocate(buf.len());
+            ep_in.submit(transfer);
+            let Some(comp) = ep_in.wait_next_complete(timeout) else {
+                return Err(std::io::ErrorKind::TimedOut.into());
+            };
```

Comment:
Note that `wait_next_complete` does not cancel the transfer when returning `None`. It will be cancelled when the `Endpoint` is dropped, but any data received between `wait_next_complete` and the cancellation being processed will be discarded. Effectively the same race as https://github.com/kevinmehall/nusb/issues/103 found in probe-rs with the prior API.

Reply:
This is a problem we also need to solve in WebUSB where we cannot cancel but we need to rely on cancelling.

Reply:
The high-level API mentioned will probably be something that manages transfers, buffers the data and implements `AsyncBufRead`. That way, cancelling an `AsyncReadExt::read` future does not have to cancel transfers at the USB level; the data just sits in the buffer until another call reads it.

Reply:
Added [EndpointRead and EndpointWrite](https://docs.rs/nusb/0.2.0-beta.2/nusb/io/index.html) in v0.2.0-beta.2